### PR TITLE
Changes for L0 framework

### PR DESCRIPTION
**Pull request by eaarora-ms**

### CODE REVIEW

File: src/Tests/azureAppServiceSettings.test.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        expect(appDetails).toHaveBeenCalled();
+    });
+
+    it('Checks valid json', async() => {
+        try {
+            validateSettings(JSON.stringify(jsonObject['connection-strings-json']));
+            validateSettings(JSON.stringify(jsonObject['app-settings-json']));
+        }
+        catch(e) { 
+        }
```
Comment (SumiranAgg) [inline src/Tests/azureAppServiceSettings.test.ts]: Can we expect the function does not error out or returns. Maybe use toHaveReturned(). Then when a error json object is added in another test case we can expect toThrowError

File: src/main.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ import { getHandler } from 'azure-actions-webclient/lib/AuthorizationHandlerFact
 
 var prefix = !!process.env.AZURE_HTTP_USER_AGENT ? `${process.env.AZURE_HTTP_USER_AGENT}` : "";
 
-async function main() {
+export default async function main() {
```
Comment (SumiranAgg) [inline src/main.ts]: Why export default and not just export.

File: src/main.ts [hunk 3]
```diff
@@ -70,14 +40,17 @@ async function main() {
         let appServiceUtility: AzureAppServiceUtility = new AzureAppServiceUtility(appService);
 
         if(AppSettings) {
+            let customApplicationSettings = validateSettings(AppSettings);
```
Comment (SumiranAgg) [inline src/main.ts]: Good refactoring. :)

**Comment by SumiranAgg** [comment]
Can you add workflow as well?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    steps:
+
+    - name: 'Checking out repo code'
+      uses: actions/checkout@v2
+
+    - name: 'Validate build'
+      run: |
+        npm install
+        npm run build
+    - name: 'Run L0 tests'
```
Comment (SumiranAgg) [inline .github/workflows/ci.yml]: add newline between steps

File: package.json [hunk 5]
```diff
@@ -5,17 +5,21 @@
   "main": "lib/main.js",
   "dependencies": {
     "@actions/core": "^1.1.3",
+    "@types/jest": "^25.1.4",
     "@types/q": "^1.5.2",
     "azure-actions-appservice-rest": "^1.0.2",
-    "azure-actions-webclient": "^1.0.1"
+    "azure-actions-webclient": "^1.0.1",
+    "ts-jest": "^25.2.1"
```
Comment (SumiranAgg) [inline package.json]: Will all jest related dependencies be in dev dependencies?

File: jest.config.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+    clearMocks: true,
+    moduleFileExtensions: ['js', 'ts'],
+    testEnvironment: 'node',
+    testMatch: ['**/*.test.ts'],
+    //testRunner: 'jest-circus/runner',
```
Comment (20shivangi) [inline jest.config.js]: Remove testrunner

**Pull request merged**